src/Nethermind/Nethermind.Merge.Gnosis/InitializationSteps/InitializeBlockchainAuRaMerge.cs

```diff
@@ -0,0 +1,83 @@
+            if (_api.StorageProvider == null) throw new StepDependencyException(nameof(_api.StorageProvider));
+            if (_api.TxPool == null) throw new StepDependencyException(nameof(_api.TxPool));
+            if (_api.ReceiptStorage == null) throw new StepDependencyException(nameof(_api.ReceiptStorage));
+            if (_api.BlockTree == null) throw new StepDependencyException(nameof(_api.BlockTree));
+            if (_api.GasPriceOracle == null) throw new StepDependencyException(nameof(_api.GasPriceOracle));
+            if (_api.ChainSpec == null) throw new StepDependencyException(nameof(_api.ChainSpec));
+
+            var processingReadOnlyTransactionProcessorSource = CreateReadOnlyTransactionProcessorSource();
+            var txPermissionFilterOnlyTxProcessorSource = CreateReadOnlyTransactionProcessorSource();
+            ITxFilter auRaTxFilter = TxAuRaFilterBuilders.CreateAuRaTxFilter(
+                _api,
+                txPermissionFilterOnlyTxProcessorSource,
+                _api.SpecProvider,
+                new ServiceTxFilter(_api.SpecProvider));
+
+            IDictionary<long, IDictionary<Address, byte[]>> rewriteBytecode = _api.ChainSpec.AuRa.RewriteBytecode;
+            ContractRewriter? contractRewriter = rewriteBytecode?.Count > 0 ? new ContractRewriter(rewriteBytecode) : null;
+
+            var processor = new AuRaMergeBlockProcessor(
+                _api.PoSSwitcher!,
+                _api.SpecProvider,
+                _api.BlockValidator,
+                _api.RewardCalculatorSource!.Get(_api.TransactionProcessor!),
+                new BlockProcessor.BlockValidationTransactionsExecutor(_api.TransactionProcessor!, _api.StateProvider!),
+                _api.StateProvider,
+                _api.StorageProvider,
+                _api.ReceiptStorage,
+                _api.LogManager,
+                _api.BlockTree,
+                auRaTxFilter,
+                GetGasLimitCalculator(),
+                contractRewriter
+            );
+
+            var auRaValidator = CreateAuRaValidator(processor, processingReadOnlyTransactionProcessorSource);
+            processor.AuRaValidator = auRaValidator;
+            var reportingValidator = auRaValidator.GetReportingValidator();
+            _api.ReportingValidator = reportingValidator;
+            if (_sealValidator != null)
+            {
+                _sealValidator.ReportingValidator = reportingValidator;
+            }
+
+            return processor;
+        }
+
+        protected override void InitSealEngine()
+        {
+            base.InitSealEngine();
+
+            if (_api.PoSSwitcher is null) throw new StepDependencyException(nameof(_api.PoSSwitcher));
+            if (_api.SealValidator is null) throw new StepDependencyException(nameof(_api.SealValidator));
+
+            _api.SealValidator = new Merge.Plugin.MergeSealValidator(_api.PoSSwitcher!, _api.SealValidator!);
+        }
```

Comment:
Didn't we want to remove InitSealEngine altogether and move base code somewhere else?

Reply:
Since I'm using InitializationSteps for everything else I moved it back here, it feels better to have everything in one place. wdyt?

Reply:
Ok lets keep it